app/mailers/user_mailer.rb

```diff
@@ -31,11 +32,12 @@ def mention_notification(user, job)
     mail(to: @user.email, subject: "#{@user.full_name}, you have been mentioned on #{@job.id}, #{@job.name}")
   end
 
-  def vendor_work_order_notification(customer, job, work_order)
+  def vendor_work_order_notification(customer, job, work_order, vendor = nil)
     @customer = customer
     @job = job
     @work_order = work_order
     subject = "#{@customer.full_name}, you have been sent a work order from Service Alliance for job,#{@job.name}. Franchise: #{@job.franchise.try(:name)}."
+    @to = vendor.nil? ? @work_order.to.join(', ') : vendor.name
```

Comment:
Again, being very picky here. 

I generally try to have less `if`'s in my code, all the code metric tools consider them as huge complexity increase ;-) I also consider 4 arguments in method as a huge code smell.

So in this case i would create two methods:
```
def work_order_notification_for_vendor(customer, job, work_order, vendor)
    ...
    @to = vendor.name
end

def work_order_notification(customer, job, work_order)
     ...
     @to = @work_order.to.join(', ')
end 
```

and then you can avoid using WorkOrderDeliveryService#deliver_vendor_email and use those two methods directly ^^

Reply:
I like this suggestion, removing unnecessary branching logic is great, thanks. :-)

Reply:
Actually @bradbergeron-us currently we're sending the vendor template to the job manager (it's missing the link back to the app, and the insurance details), is this correct or should the job manager receive the standard template?